docs/docsite/rst/reference_appendices/general_precedence.rst

```diff
@@ -1,77 +1,68 @@
 
 You can use the ``ansible-config`` command line utility to see the current value of a configuration item and where it came from.
 
+Command-line options
+--------------------
 
-.. _command_line_options_precedence:
-
-Command line options
-`````````````````````
+Any command-line option will override any configuration setting.
 
-The internal precedence is simple, but depends on the parameter used, 'last defined' wins (connects as carol, not mike)::
+At the command line, if you pass multiple values for a parameter that accepts only a single value, the 'last defined' value wins. For example, this ad-hoc task will connect as ``carol``, not as ``mike``::
 
       ansible -u mike -m ping myhost -u carol
 
-Unless the parameter allows for multiple entries and then they get appended (hosts are from inventory1 + inventory2)::
+Some parameters allow multiple values. In this case, Ansible will append all values from the hosts listed in inventory files inventory1 and inventory2)::
 
    ansible -i /path/inventory1 -i /path/inventory2 -m ping all
 
 The help for each tool specifies which parameters allow for multiple entries.
 
-Most of these options are supposed to deal with generic settings, but over time some very connection and strategy specific ones were added, they are also the main source of confusion as most of them have low precedence,
-they are mostly meant to override configuration (which does NOT include playboks, inventory nor variables).
-
-The one exception is 'extra vars' which sets variables with the highest precedence .. and since 'conneciton variables' are also the highest precedence it creates a short circut to it's own options::
-
-   ansible -u myuser -e 'ansible_user=notmyuser' -a whoami all
-
-This is the main source of confusion for people using Ansible, the above statement makes a redundant use of `-u` since it will be overriden by `ansible_user`. There is a reason for this and it has mostly to do with
-how varied different installations are, `-u` makes sense when you don't use ``ansible_user`` anywhere, it allows you to override the 'current user' or configuration entries, but once you start using variables, you need
-to keep in mind that they have high precedence making these options less useful in your context.
+Most command-line options deal with generic settings, but some settings are specific to connections and strategies.
+Passing these options at the command-line may feel like the highest-precedence options, but command-line options have low precedence - they override configuration only. They do not override playbook keywords, variables from inventory or variables from playbooks.
 
+Playbook keywords
+-----------------
 
-.. _playbook_precedence:
+Any playbook keyword will override any command-line option (but not extra variables passed at the command line with ``-e``) and any configuration setting.
```

Comment:
this is misleading, keywords override options, but not variables (command line or not)

Reply:
yeah, I see what you mean - resolved in ffec082fb09670b8a8d9b5ee88a951120972729d